src/MaterialsDesigner.jsx

```diff
@@ -17,11 +17,52 @@ import ItemsList from "./components/items_list/ItemsList";
 import SourceEditor from "./components/source_editor/SourceEditor";
 import { Material } from "./material";
 
+const GRID_CONFIG_BY_VISIBILITY = {
+    // "111" means that all three components are visible
+    "#111": {
+        1: { xs: 12, md: 2.5, lg: 2, xl: 1.5 },
+        2: { xs: 12, md: 4.75, lg: 4.375, l: 4 },
+        3: { xs: 12, md: 4.75, lg: 5.625, l: 6.5 },
+    },
+    "#001": {
+        1: { xs: 12 },
+        2: { xs: 12 },
+        3: { xs: 12 },
+    },
+    "#010": {
+        1: { xs: 12 },
+        2: { xs: 12 },
+        3: { xs: 12 },
+    },
+    "#100": {
+        1: { xs: 12 },
+        2: { xs: 12 },
+        3: { xs: 12 },
+    },
+    "#011": {
+        1: { xs: 0 },
+        2: { xs: 6 },
+        3: { xs: 6 },
+    },
+    "#101": {
+        1: { xs: 3 },
+        2: { xs: 0 },
+        3: { xs: 9 },
```

Comment:
There're some edge cases:
<img width="543" alt="image" src="https://github.com/Exabyte-io/materials-designer/assets/79542055/c06edfbc-9928-4994-8761-89f55fc392e4">
<img width="505" alt="image" src="https://github.com/Exabyte-io/materials-designer/assets/79542055/0604a69a-bae2-4ab8-8885-72740e59e052">

Maybe something like this would be better:
```
1: { xs: 12, md: 3 },
2: { xs: 0 },
3: { xs: 12, md: 9 },
```